bot/exts/valentines/markov_poem_generator.py

```diff
@@ -0,0 +1,512 @@
+            is_first_error=is_first_error
+        )
+
+        elapsed_time = datetime.now() - time_start
+        elapsed_time = elapsed_time.seconds
+
+        poem_embed = Embed(
+            title="A Markov Poem For " + str(ctx.author.name),
+            color=Colours.soft_red
+        )
+
+        await LinePaginator.paginate(
+            lines,
+            ctx,
+            poem_embed,
+            max_lines=31,
+            max_size=1000,
+            empty=False,
+            footer_text=f"Elapsed time: {elapsed_time}s\n"
+                        "Rhymes API provided by datamuse."
+        )
+
+    @commands.max_concurrency(1, commands.BucketType.user)
+    @commands.command(
+        aliases=("poem", "mpoem", "m_poem", "markpoem", "mark_poem"),
+        help=f"""This command generates a poem via a markov chain, and all you
+            have to do is provide a rhyme scheme.
+
+            The rhyme scheme is made from characters separated by slashes. E.g
+            "abab/cdcd/efef/gg". The slashes denote a new stanza, i.e, they
+            create an empty line. Same characters mean that the lines rhyme.
+            Note that the characters are case sensitive! For example, a and A
+            represent two different rhyme schemes.
+
+            You may also use our existing rhyme scheme templates:
+            {chr(10).join((k + " --- " + v) for k, v in templates.items())}""",
+        brief="Gives the user a love poem made with a markov chain."
+    )
+    async def markov_poem(
+        self,
+        ctx: commands.Context,
+        rhyme_scheme: str
+    ) -> None:
+        """
+        Gives the user a love poem based on a rhyme_scheme.
+
+        The user's `rhyme_scheme` is processed before being placed into the
+        poem generation process.
+
+        In this code, a unit is defined by a single character of the scheme. If
+        two or more units are the same, they are meant to rhyme (i.e their last
+        words rhyme).
+
+        The time taken to process the command is recorded and shown in the
+        footer of the embed.
+        """
+        scheme = rhyme_scheme.lower()
+        scheme = self.templates.get(scheme, scheme)
+        unit_count = self._get_unit_count(scheme)
+        time_start = datetime.now()
+
+        await self.send_markov_poem(ctx, scheme, unit_count, time_start)
+
+    async def cog_command_error(
+        self,
+        ctx: commands.Context,
+        error: Exception
+    ) -> None:
+        """Handles Discord errors and exceptions."""
+        if isinstance(error, RhymingSentenceNotFound):
+            error.handled = True
+            await self.send_markov_poem(
+                ctx, error.scheme, error.unit_count, error.time_start,
+                is_first_error=False
+            )
+        elif isinstance(error, RhymeAPIUnresponsive):
+            error.handled = True
+            embed = Embed(
+                title="Uh oh... Markov Poem can\'t give you a poem...",
```

Comment:
No need to escape `'`.
```suggestion
                title="Uh oh... Markov Poem can't give you a poem...",
```